software-engineering-policies/InfrastructureAsCode/terraform.md

```diff
@@ -0,0 +1,44 @@
+# Infrastructure as Code: Terraform
+
+Terraform is our tool of choice, and should be considered in the first instance.
+
+Most standards and practises at the UKHO is covered in the Training courses we provide around the use of Terraform.
+
+Videos and Presentation can be found in the **Software Engineering Team > Training Channel**.
+
+Also online resources provided by [terraform.io](https://www.terraform.io/) are very useful.
+
+## Avoid adding local modules
+
+Holding modules in the same repository as the project reduces it's audience. It is likely that you might be tempted to copy the modules you created in the previous project to your new one.
+
+Most modules can be written in a relative IP lossless way, to avoid attack vectors or concerns around security. Using public modules makes them easy to find and reuse, adhering to a standard naming comvention makes them  easy to find in GitHub, and creating releases allows for ease of specifying a version to use.
+
+Each module should be hosted it's own GitHub repo to keep the download size small. following the convention mentioned in the [template project for terraform](https://github.com/UKHO/terraform-module-template) modules. Using the terraform module template will create a repo with the contents setup as standard. Review the naming convention guide linked on the readme of the template to see how you might go about naming the module.
```

Comment:
```suggestion
Each module should be hosted it's own GitHub repo to keep the download size small, following the convention mentioned in the [template project for terraform](https://github.com/UKHO/terraform-module-template) modules. Using the terraform module template will create a repo with the contents setup as standard. Review the naming convention guide linked on the readme of the template to see how you might go about naming the module.
```
Not sure I'm convinced by this? I get keeping the download size small but it feels like that's potentially a _lot_ of tiny repos, is there a better way?